### PR TITLE
Adding initial info on linters to the style guide & readme

### PR DESCRIPTION
**Pull request by karensawrey**

### CODE REVIEW

**Comment by karensawrey** [comment]
I'd be super grateful if you could help me turn the absolute links to linter files and `help` folder in the STYLE.md into relative links, @plaindocs. I tried several variants & they all resulted in 404s.

File: styleguide/STYLE.md [hunk 1]
```diff
@@ -173,6 +173,10 @@ To add the new page to the documentation sidebar on https://buildkite.com/docs,
 `app/views/layouts/_sidebar.html.erb` with a description (for example, `"G Cloud Identity", 'integrations/sso/g-cloud-identity'` ).
 > **Note:** Ruby, which keeps the website running, interprets underscores in filenames as hyphens. So if a page is called `octopussy_cat.erb.md`, you need to add it as `octopussy-cat` to the `application.html.erb` file.
 
+### Filenames and filename linting
+Use `snake_case` for filenames. The [`.ls-lint` linter](https://github.com/buildkite/docs/blob/main/.ls-lint.yml) checks if this rule is observed.
```
Comment (plaindocs) [inline styleguide/STYLE.md:177]:
```suggestion
Use `snake_case` for `*.md.erb` files in `pages`. The [`.ls-lint` linter](https://github.com/buildkite/docs/blob/main/.ls-lint.yml) checks if this rule is observed.
```

File: styleguide/STYLE.md [hunk 2]
```diff
@@ -185,6 +189,13 @@ This is some text that you do NOT want the linter to check
 ```
 Use the `vale off` syntax before a phrase that needs to be bypassed by the linter and don't forget to turn it on again with `vale on`.
 
+### Markdown linting
+A [markdown linter](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md) is at work in Buildkite documentation.
```
Comment (plaindocs) [inline styleguide/STYLE.md:193]:
```suggestion
A [markdown linter](https://github.com/DavidAnson/markdownlint) is at work in Buildkite documentation. 
```

File: styleguide/STYLE.md [hunk 3]
```diff
@@ -185,6 +189,13 @@ This is some text that you do NOT want the linter to check
 ```
 Use the `vale off` syntax before a phrase that needs to be bypassed by the linter and don't forget to turn it on again with `vale on`.
 
+### Markdown linting
+A [markdown linter](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md) is at work in Buildkite documentation. 
+
+The enabled markdown linting rules are in [`.markdownlint.yaml`](https://github.com/buildkite/docs/blob/main/.markdownlint.yaml) file.
+
+Please keep the [help](https://github.com/buildkite/docs/tree/main/pages/agent/v3/help) files exempt from markdown linting at all times!
```
Comment (plaindocs) [inline styleguide/STYLE.md:197]:
```suggestion
```

File: styleguide/STYLE.md [hunk 4]
```diff
@@ -231,7 +242,7 @@ Use the following example to add a 'note' in the documentation.
   <h3>Setting agent defaults</h3>
   <p>Use a top-level <code>agents</code> block to <a href="/docs/pipelines/defining-steps#step-defaults">set defaults</a> for all steps in a pipeline.</p>
 </section>
-```  
+```
```
Comment (plaindocs) [inline styleguide/STYLE.md:245]:
```suggestion
```

```

**Review by plaindocs — APPROVED** [review]

**Comment by plaindocs** [comment]
I'd expect removing `https://github.com` and leaving `/buildkite/docs/blob/main/.ls-lint.yml` to work, but it really doesn't matter here as for now at least it's only viewable on GitHub.

**Pull request merged**